### ISSUE TITLE
SLF4J logger uses default log level in workers with classloader or process isolation level

### ISSUE THREAD

**Issue by alexey-tsvetkov** [gradle/gradle#8318]
### Expected Behavior

SLF4J logger in a unit of work should use a correct log level when any isolation mode is used.

### Current Behavior

SLF4J logger a unit of work always uses `LogLevel.LIFECYCLE`, 
when `IsolationMode.CLASSLOADER` or `IsolationMode.PROCESS` is used.

On the other hand `IsolationMode.NONE` works correctly. 

### Context

Until https://github.com/gradle/gradle/issues/2678 is resolved, SLF4J is the only simple alternative for logging 
(of course there can be some workarounds hacked on top of files or stdout/stderr).

### Steps to Reproduce (for bugs)

A build script (in Groovy) illustrating the issue:
```
import javax.inject.Inject
import org.slf4j.Logger
import org.slf4j.LoggerFactory

class MyWorkerTask extends DefaultTask {
    private final WorkerExecutor workerExecutor
    def IsolationMode myIsolationMode

    @Inject
    public MyWorkerTask(WorkerExecutor workerExecutor) {
        super()
        this.workerExecutor = workerExecutor
    }

    @TaskAction
    def run() {
        workerExecutor.submit(MyRunnable.class) { config ->
            config.setIsolationMode(myIsolationMode)
            config.params(myIsolationMode.toString())
        }
    }
}

class MyRunnable implements Runnable {
    private final Logger log = LoggerFactory.getLogger('MyRunnable')
    private final String isolationMode

    @Inject MyRunnable(String isolationMode) {
        this.isolationMode = isolationMode
    }

    @Override
    void run() {
        println "println from MyRunnable with $isolationMode"
        log.quiet("quiet from MyRunnable with $isolationMode")
        log.error("error from MyRunnable with $isolationMode")
        log.warn("warn from MyRunnable with $isolationMode")
        log.lifecycle("lifecycle from MyRunnable with $isolationMode")
        log.info("info from MyRunnable with $isolationMode")
        log.debug("debug from MyRunnable with $isolationMode")
        log.trace("trace from MyRunnable with $isolationMode")
    }
}

task noneIsolationModeTask(type: MyWorkerTask) {
    myIsolationMode = IsolationMode.NONE
}

task classloaderIsolationModeTask(type: MyWorkerTask) {
    myIsolationMode = IsolationMode.CLASSLOADER
}

task processIsolationModeTask(type: MyWorkerTask) {
    myIsolationMode = IsolationMode.PROCESS
}

task allTasks(dependsOn: [noneIsolationModeTask, classloaderIsolationModeTask, processIsolationModeTask])
```

I've got the following output for `./gradlew allTasks -i`:
```
> Task :noneIsolationModeTask
Task ':noneIsolationModeTask' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
println from MyRunnable with NONE
quiet from MyRunnable with NONE
error from MyRunnable with NONE
warn from MyRunnable with NONE
lifecycle from MyRunnable with NONE
info from MyRunnable with NONE
:noneIsolationModeTask (Thread[Execution worker for ':' Thread 3,5,main]) completed. Took 0.007 secs.

> Task :processIsolationModeTask
Task ':processIsolationModeTask' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
println from MyRunnable with PROCESS
quiet from MyRunnable with PROCESS
error from MyRunnable with PROCESS
warn from MyRunnable with PROCESS
lifecycle from MyRunnable with PROCESS
:processIsolationModeTask (Thread[Execution worker for ':' Thread 6,5,main]) completed. Took 0.015 secs.

> Task :classloaderIsolationModeTask
Task ':classloaderIsolationModeTask' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
println from MyRunnable with CLASSLOADER
quiet from MyRunnable with CLASSLOADER
error from MyRunnable with CLASSLOADER
warn from MyRunnable with CLASSLOADER
lifecycle from MyRunnable with CLASSLOADER
:classloaderIsolationModeTask (Thread[Execution worker for ':' Thread 2,5,main]) completed. Took 0.404 secs.
```

Note that there is `info from MyRunnable with NONE`, but neither `info from MyRunnable with PROCESS` nor `info from MyRunnable with CLASSLOADER`.

### Your Environment
 * Build scan URL: https://scans.gradle.com/s/mx3ythpcwdcos/

**Comment by alexey-tsvetkov** [comment]
I suspect the problem is that `Slf4jLoggingConfigurer#configure`, which calls `OutputEventListenerBackedLoggerContext#setLogLevel` is called only during `DaemonMain#initialiseLogging`. However, with non-NONE isolation levels `OutputEventListenerBackedLoggerContext` is loaded after `initialiseLogging` is finished.

**Comment by big-guy** [comment]
This is fixed in 5.5
